src/StateMachine.jl

```diff
@@ -139,33 +125,29 @@ function with_cases(code::CodePoint, cases::Vector{CasePartialResult})
     end
     CodePoint(cases)
 end
-function ensure_label!(code::CodePoint, state::BinderState)
-    if code.label isa Nothing
-        code.label = gensym("label", state)
-    end
-end
-function name(code::CodePoint, id::IdDict{CodePoint, Int})
-    if code.label isa Nothing
-        "State $(id[code])"
-    else
+function name(code::T, id::IdDict{T, Int}) where { T }
+    if hasfield(typeof(code), :label) && code.label !== nothing
         "State $(id[code]) ($(pretty_name(code.label)))"
+    else
+        "State $(id[code])"
     end
 end
-function successors(c::CodePoint)::Vector{CodePoint}
+function successors(c::T)::Vector{T} where { T }
```

Comment:
it doesn't seem like `T` can truly be anything here, could this be `T <: Union{...}` or similar?